dtfs-central-api/src/v1/controllers/tfm/deal/tfm-get-deals.controller.js

```diff
@@ -48,6 +48,8 @@ const sortDeals = (deals, sortBy) =>
     return 0;
   });
 
+const escapeRegExp = (string) => string.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
```

Comment:
This is a bit of a hack. 
It would probably be better to use [Text indexes ](https://www.mongodb.com/docs/manual/core/indexes/index-types/index-text/) but this would require modifying the database